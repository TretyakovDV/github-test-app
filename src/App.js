import React from 'react';
import { observer, inject } from 'mobx-react';
import './App.css';
import AddRepositoryForm from './components/AddRepositoryForm';
import RepositoryListItem from './components/RepositoryListItem';

const App = inject('repositories')(observer(({ repositories }) => {
  const { list } = repositories;

  console.log('list => ', list);

  const handleSubmit = (name) => repositories.addRepository(name);

  return (
    <div className="App">
      <header className="App-header">
        <h1>
          Mobx Test v3
        </h1>
        <AddRepositoryForm onSubmit={handleSubmit} />
        <ul>
          {
            list.length ? list.map((el) => (
              <RepositoryListItem
                key={el.id}
                title={el.title}
                stars={el.stars}
                url={el.url}
              />
            )) : (<p>List is empty!</p>)
          }
        </ul>
      </header>
    </div>
  );
}));

export default App;
