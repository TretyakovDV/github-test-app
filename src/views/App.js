import React from 'react';
import { observer, inject } from 'mobx-react';
import {
  Typography,
  Grid,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableFooter,
} from '@material-ui/core';

import AddRepositoryForm from '../components/AddRepositoryForm';
import RepositoryTableRow from '../components/RepositoryTableRow';
import Notification from '../components/Notification';
import Details from '../components/Details';

const App = inject('repositories', 'details')(observer(({ repositories, details }) => {
  const { list } = repositories;
  const handleSubmit = (name) => repositories.addRepository(name);
  const handleDelete = (id) => repositories.removeRepository(id);
  const handleDetails = (id) => details.open(id);

  return (
    <>
      <Notification />
      <Details />
      <Grid container justify="center" alignItems="center" direction="column">
        <Grid item>
          <Typography variant="h1" className="mb-20 mt-40">
            Github Stargazer
          </Typography>
        </Grid>
        <Grid item>
          <AddRepositoryForm onSubmit={handleSubmit} />
        </Grid>
        <Grid item>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell width={250}>Name</TableCell>
                <TableCell width={100} align="right">Stars</TableCell>
                <TableCell align="center">Details</TableCell>
                <TableCell align="center">Link</TableCell>
                <TableCell align="center">Delete</TableCell>
              </TableRow>
            </TableHead>

            {list.length ? (
              <TableBody>
                {list.map((el) => (
                  <RepositoryTableRow
                    key={el.id}
                    id={el.id}
                    title={el.title}
                    url={el.url}
                    stars={el.stars}
                    onDelete={handleDelete}
                    onDetails={handleDetails}
                  />
                ))}
              </TableBody>
            ) : (
              <TableFooter>
                <TableRow>
                  <TableCell align="center" variant="footer" colSpan={5}>
                    <Typography variant="h4">List is empty</Typography>
                  </TableCell>
                </TableRow>
              </TableFooter>
            )}
          </Table>
        </Grid>
      </Grid>
    </>
  );
}));

export default App;
