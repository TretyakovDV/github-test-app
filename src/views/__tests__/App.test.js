import React from 'react';
import renderWithProvider from '../../utils/test/renderWithProvider';
import App from '../App';

describe('App', () => {
  const defaultStore = {
    repositories: {
      getRepository: jest.fn(),
      removeRepository: jest.fn(),
      list: [
        {
          id: 1,
          title: 'title',
          stars: 1,
          url: 'url',
        },
      ],
      languages: jest.fn(),
    },
    details: {
      open: jest.fn(),
    },
    error: {},
  };

  describe('with repositories', () => {
    it('matches snapshot', () => {
      const { asFragment } = renderWithProvider(<App />, defaultStore);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('without repositories', () => {
    const store = {
      ...defaultStore,
      repositories: {
        ...defaultStore.repositories,
        list: jest.fn(() => ([])),
      },
    };

    it('matches snapshot', () => {
      const { asFragment } = renderWithProvider(<App />, store);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
