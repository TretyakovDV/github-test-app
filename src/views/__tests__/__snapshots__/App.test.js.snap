// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App with repositories matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h1
        class="MuiTypography-root mb-20 mt-40 MuiTypography-h1"
      >
        Github Stargazer
      </h1>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root paper mb-20 MuiPaper-elevation10 MuiPaper-rounded"
      >
        <div
          class="MuiFormControl-root MuiTextField-root paper__input"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              placeholder="facebook/react"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root add-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
              width="250"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
              width="100"
            >
              Stars
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Details
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Link
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Delete
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
              width="250"
            >
              title
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              width="100"
            >
              1
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="details"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14 17H4v2h10v-2zm6-8H4v2h16V9zM4 15h16v-2H4v2zM4 5v2h16V5H4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root"
                href="url"
                tabindex="0"
                target="_blank"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="delete"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App without repositories matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <h1
        class="MuiTypography-root mb-20 mt-40 MuiTypography-h1"
      >
        Github Stargazer
      </h1>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiPaper-root paper mb-20 MuiPaper-elevation10 MuiPaper-rounded"
      >
        <div
          class="MuiFormControl-root MuiTextField-root paper__input"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              placeholder="facebook/react"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiIconButton-root add-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
              width="250"
            >
              Name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
              width="100"
            >
              Stars
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Details
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Link
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              Delete
            </th>
          </tr>
        </thead>
        <tfoot
          class="MuiTableFooter-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-footer"
          >
            <td
              class="MuiTableCell-root MuiTableCell-footer MuiTableCell-alignCenter"
              colspan="5"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4"
              >
                List is empty
              </h4>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
